### ISSUE TITLE
Cannot build develop branch due to `unreachable code` in `ScMap.hpp`

### ISSUE THREAD

**Issue by seifer7** [OpenRCT2/OpenRCT2#15228]
Since commit# 1168425f14ffeebdabb9624702e18b78fde96025 I've been unable to build the develop branch without removing the newly added line:
https://github.com/OpenRCT2/OpenRCT2/blob/1168425f14ffeebdabb9624702e18b78fde96025/src/openrct2/scripting/ScMap.hpp#L240

### Error encountered:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2220	the following warning is treated as an error	libopenrct2	\src\openrct2\scripting\ScMap.hpp	240	
Warning	C4702	unreachable code	libopenrct2	\src\openrct2\scripting\ScMap.hpp	240	
```

### Build attempt using
Windows 10, 64-bit, Version 10.0.19044 Build 19044
Using Microsoft Visual Studio 2019 Version 16.10.4

**Comment by Gymnasiast** [comment]
The line was added because on my system (using GCC 9), it wouldn’t build _without_ it :-/

What happens if you replace the line with `assert(false);`? Does it build then?

**Comment by duncanspumpkin** [comment]
Suggest doing:
```
#pragma warning( push )
#pragma warning( disable : 4702 )
return DukValue(); 
#pragma warning( pop )
```

**Comment by Broxzier** [comment]
Building the game with Microsoft Visual Studio Version 16.11.0 (Preview) I do not get this error. Going with @duncanspumpkin's proposal, it should be mentioned in a comment why it's necessary. Once the current Preview version becomes the next stable build, I think they should be removed again.

**Comment by IntelOrca** [comment]
Not sure if this is fixed in #15186, but this can be solved by just only have one return statement at the end of the function. Use single entry, single exit pattern.
